### PR TITLE
prompt collections

### PR DESCRIPTION
**Pull request by hwchase17**

### CODE REVIEW

File: langchain/src/chains/prompt_collections.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+export abstract class BasePromptCollection {
+  abstract getPrompt(llm: BaseLanguageModel): BasePromptTemplate;
+}
+
+export class ConditionalPromptCollection extends BasePromptCollection {
+  defaultPrompt: BasePromptTemplate;
+
+  conditionals: Array<
+    [condition: (llm: BaseLanguageModel) => boolean, prompt: BasePromptTemplate]
+  >;
```
Comment (agola11) [inline langchain/src/chains/prompt_collections.ts:15]: nit: make this a type

**Review by agola11 — APPROVED** [review]
looks good, one small comment

File: langchain/src/chains/question_answering/load.ts [hunk 2]
```diff
@@ -44,3 +46,43 @@ export const loadQAChain = (llm: BaseLLM, params: qaChainParams = {}) => {
   }
   throw new Error(`Invalid _type: ${type}`);
 };
+
+interface stuffQAChainParams {
```
Comment (nfcampos) [inline langchain/src/chains/question_answering/load.ts:50]: Types should start with uppercase letter

File: langchain/src/chains/question_answering/load.ts [hunk 3]
```diff
@@ -44,3 +46,43 @@ export const loadQAChain = (llm: BaseLLM, params: qaChainParams = {}) => {
...
+  llm: BaseLLM,
+  params: stuffQAChainParams = {}
+) => {
+  const { prompt = QA_PROMPT_SELECTOR.getPrompt(llm) } = params;
+  const llmChain = new LLMChain({ prompt, llm });
+  const chain = new StuffDocumentsChain({ llmChain });
+  return chain;
+};
+
+interface mapReduceQAChainParams {
```
Comment (nfcampos) [inline langchain/src/chains/question_answering/load.ts:64]: Same as above

**Review by nfcampos — COMMENTED** [review]
Lgtm

**Pull request merged**